File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        max_area = 0
        
        start = 0
        end = len(height) - 1
        
        while start < end:
            
            area = (end-start) * min(height[end], height[start])
            max_area = max(max_area, area)
            
            if height[end] > height[start]:
                start += 1
            else:
                end -= 1
            
        return max_area

File: 11-container-with-most-water/11-container-with-most-water.go
func maxArea(height []int) int {
	n := len(height)

	var l, result int
	r := n - 1

	for l < r {
		currArea := min(height[l], height[r]) * (r - l)
		result = max(result, currArea)
        
		if height[l] < height[r] {
			l++
		} else {
			r--
		}
	}
	return result
}

func min(a,b int) int {
    if a<b{
        return a
    }
    return b
}

func max(a, b int) int {
    if a>b {
        return a
    }
    return b
}